rust/datafusion/src/sql/planner.rs

```diff
@@ -421,16 +415,85 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         // The SELECT expressions, with wildcards expanded.
         let select_exprs = self.prepare_select_exprs(&plan, &select.projection)?;
 
+        // Optionally the HAVING expression.
+        let having_expr_opt = if let Some(having_expr) = &select.having {
```

Comment:
This probably could use `Option::map` instead of `if let`

Reply:
Updated in c04743c.